src/arch/metal.rs

```diff
@@ -621,27 +622,61 @@ _start:
     add t0, t0, t3       // The actual start of our stack
     add t1, t0, t1       // And the end of our stack
 
+    ld t4, __bss_start
+    ld t5, __bss_stop
+
     // Then we fill the stack with a known memory pattern
     li t2, 0x0BADBED0
+
 loop:
-    bgeu t0, t1, done // Exit when reaching the end address
+    bgeu t0, t1, zero_loop // Exit when reaching the end address
     sw t2, 0(t0)      // Write the pattern
     addi t0, t0, 4    // increment the cursor
     j loop
-done:
 
+zero_loop:
+    bgeu t4, t5, verify
+    sb x0, 0(t4)
+    addi t4, t4, 1
+    j zero_loop
+
+verify:
+    ld t4, __bss_start
+    ld t5, __bss_stop
+    j verify_loop
```

Comment:
You don't need that last jump, because the PC will advance to the next instruction which happens to be `verify_loop` after the last instruction of the `verify` block.

```suggestion
```